### PR TITLE
Added support of async modal for the button field

### PR DESCRIPTION
**Pull request by Nks**
I think we need loadModalAsync for the button because sometimes you need to send data to the modal but if we using async modal it overriding attributes which should be sent with the params.

### CODE REVIEW

File: resources/views/fields/button.blade.php [hunk 1]
```diff
@@ -3,6 +3,8 @@
         <button type="button"
                 class="{{ $class }}"
                 data-action="screen--base#targetModal"
+                data-async="{{ ($async ? 'true' : 'false') }}"
```
Comment (tabuna) [inline resources/views/fields/button.blade.php]: Strict types allow you to make functions like this:

```
data-async="{{ var_export($async) }}"
```

File: resources/views/fields/button.blade.php [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
+                @if($async)data-modal-params="@json($async_params)"@endif
```
Comment (tabuna) [inline resources/views/fields/button.blade.php]: Let's not write in style `snake case`

File: src/Screen/Fields/Button.php [hunk 3]
```diff
@@ -124,9 +126,20 @@ public function type(string $visual): self
...
+        ], '', (string)$this->get('class'));
+
+        $this->set('class', $class . ' ' . $visual);
+
+        return $this;
+    }
 
-        $this->set('class', $class.' '.$visual);
+
+    public function loadModalAsync(string $modal, $method, $options, string $text = null): self
```
Comment (tabuna) [inline src/Screen/Fields/Button.php]: Let's try to keep a single name in the executable file and the template file.
Also the absence of phpdocs makes the whole method look.


Let's write it more strictly, for example, options are declared by default as an array, 
why don't we allow them as `array $options = []`

File: resources/views/fields/button.blade.php (hunk 1 above)
Comment (Nks) [reply resources/views/fields/button.blade.php]: Fixed

File: src/Screen/Fields/Button.php (hunk 3 above)
Comment (Nks) [reply src/Screen/Fields/Button.php]: Added phpdoc and wrapper

File: resources/views/fields/button.blade.php (hunk 2 above)
Comment (Nks) [reply resources/views/fields/button.blade.php]: Rewrite in camelCase

**Pull request merged**